### PR TITLE
Style Variations: Fix the fonts are not displayed correctly

### PR DESCRIPTION
**Pull request by arthur791004**
Related to D132838-code

## Proposed Changes

* Fix the fonts on the style variations are not displayed correctly because the inline CSS doesn't include the font definitions.

## Testing Instructions

* Apply D132838-code to your sandbox
* Go to the Theme Showcase or the Design Picker
* Pick the theme with the style variations, e.g.: TT4, Zaino
* Make sure the style variations are displayed correctly

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by taipeicoder** [comment]
I'm seeing some 404 errors related to fonts:

https://github.com/Automattic/wp-calypso/assets/797888/5e8b267b-72de-4cfb-bda5-843cacde8253

@arthur791004 are you able to reproduce it? For more context, I have the patch D132838-code applied.

File: packages/global-styles/src/components/global-styles-variation-container/index.tsx [hunk 1]
```diff
@@ -74,6 +79,9 @@ const GlobalStylesVariationContainer = ( {
 			<EditorStyles styles={ editorStyles ?? [] } />
 			{ containerResizeListener }
 			{ children }
+			{ fontFamilies && fontFamilies.length > 0 && (
+				<FontFamiliesLoader fontFamilies={ fontFamilies } onLoad={ onFontFamiliesLoad } />
```
Comment (fushar) [inline packages/global-styles/src/components/global-styles-variation-container/index.tsx]: Why do we need to load the fonts here? I believe, every theme should include ALL the fonts referenced by the style variations, even if they are Jetpack Google fonts. So all fonts must have been printed by D132838-code.

Or could you point to a theme (+ variation) which won't work if we don't load it via Google Fonts here?

**Comment by fushar** [comment]
> I'm seeing some 404 errors related to fonts:

I think this is because Podcasty's h1 is using Secuela font, which is not available as Google Font, thus the error.

See also my comment here: https://github.com/Automattic/wp-calypso/pull/85641#discussion_r1434663728

**Comment by fushar** [comment]
AFAIK, the original intention of the `<FontFamiliesLoader />` is to load the Google Fonts for the Creatio 2 theme in the Assembler flow, because the font pairings are Jetpack Google Fonts, not the ones that are included in Creatio 2.

**Comment by arthur791004** [comment]
Closing it as D132838-code should resolve this issue 🙂

**Comment by arthur791004** [comment]
Oh, we still need this PR to fall back to the heading font if the h1 font is not defined.

**Comment by taipeicoder** [comment]
> Oh, we still need this PR to fall back to the heading font if the h1 font is not defined.

Could you provide some themes where the h1 font is not defined? Edit: Zaino, it seems 🙂

**Comment by arthur791004** [comment]
Yes, Zaino 😆

**Review by taipeicoder — APPROVED** [review]
Tested with Zaino, and confirmed that the font is Archivo as defined in the theme.json fbhepr%2Skers%2Sjcpbz%2Qcerzvhz%2Qgurzrf%2Smnvab%2Sgurzr.wfba%3Se%3Q699r351p%23690-og

Screenshot 2023-12-25 at 4 01 03 PM

**Pull request merged**